packages/core/bootstrap/src/lib/config/provider-limits/config.ts

```diff
@@ -30,8 +36,24 @@ export function get(
   const perMinuteRateLimit = getEnv('RATE_LIMIT_CAPACITY_MINUTE')
   const shouldIgnorePerSecLimit = perSecRateLimit && parseInt(perSecRateLimit) <= 0
   const shouldIgnorePerMinLimit = perMinuteRateLimit && parseInt(perMinuteRateLimit) <= 0
-  if (perSecRateLimit) capacity = shouldIgnorePerSecLimit ? 0 : parseInt(perSecRateLimit)
-  if (perMinuteRateLimit) capacity = shouldIgnorePerMinLimit ? 0 : parseInt(perMinuteRateLimit)
+  let highestTierLimit = 0
+
+  if (perSecRateLimit) {
+    capacity = shouldIgnorePerSecLimit ? 0 : parseInt(perSecRateLimit)
+    highestTierLimit = getLastTierLimitValue(rateLimitConfig.limits, 'http', 'rateLimit1s')
+  }
+
+  if (perMinuteRateLimit) {
+    capacity = shouldIgnorePerMinLimit ? 0 : parseInt(perMinuteRateLimit)
+    highestTierLimit = getLastTierLimitValue(rateLimitConfig.limits, 'http', 'rateLimit1m')
+  }
+
+  if (enabled && capacity > highestTierLimit) {
+    logger.warn(
+      `The configured RATE_LIMIT_CAPACITY value is higher than the highest tier value from limits.json`,
```

Comment:
To make this message as descriptive as possible for NOPs, we should maybe mention `RATE_LIMIT_CAPACITY_SECOND` and `RATE_LIMIT_CAPACITY_MINUTE` in here too. Since custom limits can be set through those env vars without `RATE_LIMIT_CAPACITY` being set.